packages/nimble-components/src/menu-button/tests/menu-button.spec.ts

```diff
@@ -256,86 +227,92 @@ describe('MenuButton', () => {
 
         it("should fire 'toggle' event when the menu is opened", async () => {
             await connect();
-            const toggleListener = createEventListener(element, 'toggle');
+            const spy = jasmine.createSpy<MenuButtonToggleEventHandler>();
+            const toggleListener = waitForEvent(element, 'toggle', spy);
             element.open = true;
-            await toggleListener.promise;
-            expect(toggleListener.spy).toHaveBeenCalledTimes(1);
+            await toggleListener;
+            expect(spy).toHaveBeenCalledTimes(1);
             const expectedDetails: MenuButtonToggleEventDetail = {
                 newState: true,
                 oldState: false
             };
-            const event = toggleListener.spy.calls.first()
-                .args[0] as CustomEvent;
+            const event = spy.calls.first().args[0];
             expect(event.detail).toEqual(expectedDetails);
         });
 
         it("should fire 'toggle' event when the menu is closed", async () => {
             element.open = true;
             await connect();
-            const toggleListener = createEventListener(element, 'toggle');
+            const spy = jasmine.createSpy<MenuButtonToggleEventHandler>();
+            const toggleListener = waitForEvent(element, 'toggle', spy);
             element.open = false;
-            await toggleListener.promise;
-            expect(toggleListener.spy).toHaveBeenCalledTimes(1);
+            await toggleListener;
+            expect(spy).toHaveBeenCalledTimes(1);
             const expectedDetails: MenuButtonToggleEventDetail = {
                 newState: false,
                 oldState: true
             };
-            const event = toggleListener.spy.calls.first()
-                .args[0] as CustomEvent;
+            const event = spy.calls.first().args[0];
             expect(event.detail).toEqual(expectedDetails);
         });
 
         it("should fire 'beforetoggle' event before the menu opens", async () => {
             await connect();
-            const toggleListener = createEventListener(element, 'toggle');
-            const beforeToggleListener = createBeforeToggleListener(
+            const spy = jasmine.createSpy<MenuButtonToggleEventHandler>();
```

Comment:
This ends up working nice since both events return the same type. Otherwise a shared spy would need something more generic like or some union type, etc.

```
type CustomEventHandler = (
    evt: CustomEvent
) => void;
```